x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/response/HuggingFaceRerankResponseEntity.java

```diff
@@ -0,0 +1,130 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.huggingface.response;
+
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.core.inference.results.RankedDocsResults;
+import org.elasticsearch.xpack.inference.external.http.HttpResult;
+import org.elasticsearch.xpack.inference.services.huggingface.request.rerank.HuggingFaceRerankRequest;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.elasticsearch.common.xcontent.XContentParserUtils.parseList;
+import static org.elasticsearch.core.Strings.format;
+import static org.elasticsearch.xpack.inference.external.response.XContentUtils.moveToFirstToken;
+
+public class HuggingFaceRerankResponseEntity {
+
+    private static final String FAILED_TO_FIND_FIELD_TEMPLATE = "Failed to find required field [%s] in Hugging Face rerank response";
+    private static final String INVALID_ID_FIELD_FORMAT_TEMPLATE = "Expected numeric value for record ID field in Hugging Face rerank ";
+
+    /**
+     * Parses the Hugging Face rerank response.
+
+     * For a request like:
+     *
+     * <pre>
+     *     <code>
+     *         {
+     *              "input": ["luke", "like", "leia", "chewy","r2d2", "star", "wars"],
+     *              "query": "star wars main character",
+     *              "return_documents": true,
+     *              "top_n": 1
+     *          }
+     *     </code>
+     * </pre>
+
+     * The response would look like:
+
+     * <pre>
+     *     <code>
+     *         {
+     *              "rerank": [
+     *                  {
+     *                       "index": 5,
+     *                       "relevance_score": -0.06920313,
+     *                       "text": "star"
+     *                   }
+     *               ]
+     *          }
+     *     </code>
+     * </pre>
+     */
+
+    public static RankedDocsResults fromResponse(HuggingFaceRerankRequest request, HttpResult response) throws IOException {
+        var parserConfig = XContentParserConfiguration.EMPTY.withDeprecationHandler(LoggingDeprecationHandler.INSTANCE);
+
+        try (XContentParser jsonParser = XContentFactory.xContent(XContentType.JSON).createParser(parserConfig, response.body())) {
+            moveToFirstToken(jsonParser);
+
+            XContentParser.Token token = jsonParser.currentToken();
```

Comment:
I think we can omit this line and the `ensureExpectedToken` because the `parseList` will do the same check.